src/main/java/org/kainos/ea/db/AuthDao.java

```diff
@@ -0,0 +1,127 @@
+package org.kainos.ea.db;
+
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.MalformedJwtException;
+import org.apache.commons.lang3.time.DateUtils;
+import org.kainos.ea.cli.Credential;
+import org.kainos.ea.client.ActionFailedException;
+import org.kainos.ea.client.AuthenticationException;
+import org.springframework.security.crypto.argon2.Argon2PasswordEncoder;
+
+import javax.crypto.spec.SecretKeySpec;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.Base64;
+import java.util.Date;
+import java.security.Key;
+
+
+public class AuthDao {
+
+    private final Key hmacKey;
+    Argon2PasswordEncoder encoder = new Argon2PasswordEncoder(32, 64, 1, 15 * 1024, 2);
+
+    public AuthDao() throws ActionFailedException {
+        try {
+            hmacKey = new SecretKeySpec(Base64.getDecoder().decode(System.getenv("JWT_SECRET")),
+                    SignatureAlgorithm.HS256.getJcaName());
+        } catch (Exception e) {
+            throw new ActionFailedException("JWT_SECRET not set up correctly");
+        }
+    }
+
+    public void registerUser(Connection c, Credential credential) throws ActionFailedException {
+        String encodedPassword = encoder.encode(credential.getPassword());
+
+        String insertQuery = "INSERT INTO user(username,password) VALUES(?,?)";
+
+        try(PreparedStatement st = c.prepareStatement(insertQuery, Statement.RETURN_GENERATED_KEYS)) {
+            st.setString(1, credential.getUsername());
+            st.setString(2, encodedPassword);
+
+            int affectedRows = st.executeUpdate();
+
+            if (affectedRows == 0) {
+                throw new ActionFailedException("SQL Error: Creating user failed, no rows affected.");
+            }
+        } catch (SQLException e) {
+            System.err.println(e.getMessage());
+            throw new ActionFailedException("SQL Error: User could be registered");
+        }
+    }
+
+    public boolean validateLogin(Connection c, Credential credential) throws ActionFailedException {
+        String selectQuery = "SELECT password FROM `user` WHERE username = ?";
+
+        try(PreparedStatement st = c.prepareStatement(selectQuery)) {
+            st.setString(1, credential.getUsername());
+
+            ResultSet rs = st.executeQuery();
+
+            if (rs.next()) {
+                String storedEncodedPassword = rs.getString("password");
+                return encoder.matches(credential.getPassword(), storedEncodedPassword);
+            }
+            return false;
+        } catch (SQLException e) {
+            System.err.println(e.getMessage());
+            throw new ActionFailedException("SQL Error: Login could not be validated");
```

Comment:
This message can make its way back up to the user, that probably isn't a good thing

![image](https://github.com/steveh2707/IWantItDataWay-API/assets/41385872/7baea4b9-670b-476d-9664-926a7f1c7622)